src/main.cpp

```diff
@@ -1317,6 +1324,15 @@ MempoolReturnValue AcceptCertificateToMemoryPool(CTxMemPool& pool, CValidationSt
                 }
             }
 
+            // checking txs commitment tree validity
+            SidechainTxsCommitmentBuilder scCommitmentBuilder;
+            if (!scCommitmentBuilder.add(cert, view)) {
+                state.Invalid(
```

Comment:
I have two questions here:

1. shouldn't this piece of code be inside `CheckCertificate` rather than in `AcceptCertificateToMemoryPool`? in this way the code would be more aligned with transaction analogous (`AcceptTxToMemoryPool`) and even more important the check would involve both mempool-validation as well as block-validation

2. why in this case failure results in a call to `state.Invalid` while on the tx analogous failure results in a call to `state.DoS`?

Reply:
1. See one of the previous answer
2. Yes, agree. Changed back to `DoS` in e6db47d33